prime-router/metadata/hl7_mapping/receivers/STLTs/CA/ca-specimen.yml

```diff
@@ -53,32 +53,32 @@ elements:
     hl7Spec: [ '%{hl7SpecimenFieldPath}-4-7' ]
 
   - name: ca-specimen-collection-time
-    condition: '%resource.collection.collected is dateTime'
+    condition: '%resource.collection.collected.exists() and %resource.collection.collected.is(dateTime)'
     value: [ '%resource.collection.collected.changeTimezone(%timezone)' ]
     hl7Spec: [ '%{hl7SpecimenFieldPath}-17-1' ]
 
   - name: specimen-observation-date-time-start
-    condition: '%resource.collection.collected is Period'
+    condition: '%resource.collection.collected.exists() and %resource.collection.collected is Period'
     value: [ '%resource.collection.collected.start.changeTimezone(%timezone)' ]
     hl7Spec: [ '%{hl7SpecimenFieldPath}-17-1' ]
 
   # Alternative to getting specimen collection time from DiagnosticReport instead
   - name: specimen-collection-time-diagnostic
-    condition: '%resource.collection.collected.exists().not() and %diagnostic.effective is dateTime'
+    condition: '%resource.collection.collected.exists().not() and %diagnostic.effective.exists() and %diagnostic.effective is dateTime'
     value: [ '%diagnostic.effective.changeTimezone(%timezone)' ]
     hl7Spec: [ '%{hl7SpecimenFieldPath}-17-1' ]
 
   - name: specimen-observation-date-time-end
-    condition: '%resource.collection.collected is Period'
+    condition: '%resource.collection.collected.exists() and %resource.collection.collected is Period'
     value: [ '%resource.collection.collected.end.changeTimezone(%timezone)' ]
     hl7Spec: [ '%{hl7SpecimenFieldPath}-17-2' ]
 
   - name: specimen-received-time
-    condition: '%resource.receivedTime is dateTime'
+    condition: '%resource.receivedTime.exists() and %resource.receivedTime is dateTime'
     value: [ '%resource.receivedTime.changeTimezone(%timezone)' ]
     hl7Spec: [ '%{hl7SpecimenFieldPath}-18' ]
 
   - name: specimen-received-time-diagnostic
-    condition: '%resource.receivedTime.exists().not() and %resource.collection.collected is dateTime'
+    condition: '%resource.receivedTime.exists().not() and %resource.collection.collected.is(dateTime)'
```

Comment:
Was this change meaningful?

Reply:
It was not.  I'll switch it back.  I think this is a good topic to roll into formatting too  to write that consistently